pkg/lib/cockpit.js

```diff
@@ -2090,6 +2090,10 @@ function factory() {
                 path,
                 tag: expected_tag
             };
+
+            if (attrs)
+                opts.attrs = attrs;
```

Comment:
No capability check here at least?

Reply:
So that would check if `cockpit.info?.channels?.fsreplace1?.attrs` and if not raise?

Reply:
I think this is kinda useless, we already have capabilities in 336 since:

```
[jelle@carbon][~/projects/cockpit/fsreplace1-chown]%git tag --contains ab9bb934d9dfbbaa220eec03d3bd6e73b69e4630
336
```

Which gives us nothing because it will just throw for passing `attrs` until we merge the code here. So I am having second thoughts about this whole capabilities thing it is just more work

Reply:
Thankfully it also throws a `not-supported` in the case of missing attrs.

Reply:
To proof this whole thing I have just implemented setting `mode` to this PR and started using it in Cockpit. Both `cockpit.info` and fsreplace1 give `not-supported` so it is sort of useful as fallback:

```
{
  "message": "attrs must be empty",
  "problem": "not-supported",
  "command": "close",
  "channel": "ch1"
}
```

New things would need to handle `protocol-error` as a new `attrs` option might not be supported.

```
{
  "message": "\"attrs\" contains unsupported key(s) ['banana']",
  "unsupported-attrs": [
    "banana"
  ],
  "problem": "protocol-error",
  "command": "close",
  "channel": "ch1"
}
```

Reply:
See this [commit](https://github.com/cockpit-project/cockpit/pull/21128/commits/adbb1465a633734dcb818881125e09454dcfce1f#diff-9e1ea4bddf6a6748a81b5ddce21acf44f9e54ab085d1897693854f820b3dada8R311)